imageroot/actions/restore-module/06copyenv

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+import os
+import sys
+import json
+import agent
+
+request = json.load(sys.stdin)
+
+original_environment = request['environment']
+
+for evar in [
+        "TRAEFIK_HOST",
+        "TRAEFIK_HTTP2HTTPS",
+        "TRAEFIK_LETS_ENCRYPT",
+        "IMAP_HOST"
+    ]:
+    agent.set_env(evar, original_environment[evar])
+
+# web uses the first one, archive uses the second random port
+env_tcp_ports = os.environ["TCP_PORTS"].split(',')
+tcp_port_web = env_tcp_ports[0]
+tcp_port_archive = env_tcp_ports[1]
+
+# Talk with agent using file descriptor.
+# Setup configuration from user input.
+agent.set_env("TCP_PORT_WEB", tcp_port_web )
+agent.set_env("TCP_PORT_ARCHIVE", tcp_port_archive)
```

Comment:
I'm not sure about this: should we restore also the TCP ports? Or ask the system to allocate 2 new ports?
@DavidePrincipi what do you think?

Reply:
No, TCP ports has not to be restored.

When a module instance is restored (or cloned) it is like new module creation: it must ask the core the required port numbers to avoid port conflicts with other instances.

If necessary, the port numbers should be notified to other modules so they can update their configuration.

Reply:
Relative to the port notification I think it is on the roadmap, we need to store inside redis which node runs an archive instance, what hostname to reach it and which port is listening SMTP  connexions. What do you think @DavidePrincipi

Reply:
I'd discuss this point in another PR, if you agree.